webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/queries/LdapQueryProcessor.scala

```diff
@@ -513,7 +513,10 @@ class InternalLDAPQueryProcessor(
                       res
                     }
                   }
-      postFiltered = postFilterNode(inverted.distinctBy(_.dn), query.returnType, limitToNodeIds)
+      // distinctBy computes the hashcode of the object
+      // It is really expensive on LDAP entries.
+      // The dn string is already computed, so the toString is a good alternative (and as also unicity)
```

Comment:
has also

Reply:
ok, so the cost is because `.dn.hashcode` does `.dn.toNormalizeString.hashcode` and :
- we have normalize string by construction in rudder 
- but `toNormalizeString` is extremely costly: it goes through each `rdn`, check with schema, normalize it, and then hashcode 
- and we already have the string dn (always present in ldap entries)